src/realm/table.cpp

```diff
@@ -2536,8 +2536,68 @@ void Table::update_from_parent(size_t old_baseline) noexcept
     m_alloc.bump_storage_version();
 }
 
+void Table::schema_to_json(std::ostream& out, const std::map<std::string, std::string>& renames) const
+{
+    out << "{";
+    auto name = get_name();
+    if (renames.count(name))
+        name = renames.at(name);
+    out << "\"name\":\"" << name << "\"";
+    if (this->m_primary_key_col) {
+        out << ",";
+        out << "\"primaryKey\":\"" << this->get_column_name(m_primary_key_col) << "\"";
+    }
```

Comment:
If this were sync specific you could probably make this and else since not having a primary key means that the schema is automatically embedded, but I guess this tool is also used for non-sync use cases so LGTM

Reply:
Yes - I would like to make as few assumptions as possible here.